src/battle_util.c

```diff
@@ -8011,7 +8028,8 @@ bool32 IsMoveMakingContact(u16 move, u8 battlerAtk)
         else
             return FALSE;
     }
-    else if (GetBattlerAbility(battlerAtk) == ABILITY_LONG_REACH)
+    else if (GetBattlerAbility(battlerAtk) == ABILITY_LONG_REACH
+        || GetBattlerHoldEffect(battlerAtk, TRUE) == HOLD_EFFECT_PUNCHING_GLOVE)
```

Comment:
`GetBattlerHoldEffect` is being checked twice, gotta save it in a variable.

Reply:
pretty sure the compiler would optimize this

Reply:
you sure? I wouldn't think that it would keep that in memory

Reply:
I checked the compiler explorer and you're right. Just fixed this and loaded dice.